src/Disks/IDiskRemote.cpp

```diff
@@ -482,45 +483,61 @@ void IDiskRemote::replaceFile(const String & from_path, const String & to_path)
 
 void IDiskRemote::removeSharedFile(const String & path, bool delete_metadata_only)
 {
-    std::vector<String> paths_to_remove;
+    std::unordered_map<String, std::vector<String>> paths_to_remove;
     removeMetadata(path, paths_to_remove);
 
     if (!delete_metadata_only)
-        removeFromRemoteFS(paths_to_remove);
+        removeFromRemoteFS(paths_to_remove[path]);
 }
 
 void IDiskRemote::removeSharedFileIfExists(const String & path, bool delete_metadata_only)
 {
-    std::vector<String> paths_to_remove;
+    std::unordered_map<String, std::vector<String>> paths_to_remove;
     if (metadata_disk->exists(path))
     {
         removeMetadata(path, paths_to_remove);
         if (!delete_metadata_only)
-            removeFromRemoteFS(paths_to_remove);
+            removeFromRemoteFS(paths_to_remove[path]);
     }
 }
 
-void IDiskRemote::removeSharedFiles(const RemoveBatchRequest & files, bool delete_metadata_only)
+void IDiskRemote::removeSharedFiles(const RemoveBatchRequest & files, bool keep_all_batch_metadata, const NameSet & file_names_remove_metadata_only)
 {
-    std::vector<String> paths_to_remove;
+    std::unordered_map<String, std::vector<String>> paths_to_remove;
     for (const auto & file : files)
     {
         bool skip = file.if_exists && !metadata_disk->exists(file.path);
         if (!skip)
             removeMetadata(file.path, paths_to_remove);
     }
 
-    if (!delete_metadata_only)
-        removeFromRemoteFS(paths_to_remove);
+    if (!keep_all_batch_metadata)
```

Comment:
Prev name `delete_metadata_only` is more readable to me ...

Reply:
Maybe it should be like `keep_remote_data`?

Reply:
Done.